pkg/client/metadata.go

```diff
@@ -0,0 +1,57 @@
+/*
+ * @license
+ * Copyright 2023 Dynatrace LLC
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package client
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/dynatrace/dynatrace-configuration-as-code/pkg/rest"
+	"net/http"
+)
+
+const classicEnvironmentDomainPath = "/platform/core/v1/environment-api-info" // TODO: once available, change this to /platform/metadata/v1/classic-environment-domain
+
+type classicEnvURL struct {
+	Endpoint string `json:"endpoint"`
+}
+
+// GetDynatraceClassicURL tries to fetch the URL of the classic environment using the API of a platform enabled
+// environment
+func GetDynatraceClassicURL(client *http.Client, environmentURL string) (string, error) {
+	endpointURL := environmentURL + classicEnvironmentDomainPath
+
+	resp, err := rest.Get(client, endpointURL)
+	if err != nil {
+		return "", fmt.Errorf("failed to query classic environment url %w", err)
+	}
+
+	if resp.StatusCode != http.StatusOK {
```

Comment:
```suggestion
	if !resp.IsSuccess() {
```

would cover all 2xx codes